Yosemite/YosemiteTests/Stores/ProductStoreTests.swift

```diff
@@ -550,7 +550,7 @@ final class ProductStoreTests: XCTestCase {
 
         productStore.upsertStoredProduct(readOnlyProduct: sampleProduct(), in: viewStorage)
         XCTAssertEqual(viewStorage.countObjects(ofType: Storage.Product.self), 1)
-        XCTAssertEqual(viewStorage.countObjects(ofType: Storage.ProductTag.self), 9)
+        XCTAssertEqual(viewStorage.countObjects(ofType: Storage.ProductTag.self, matching: NSPredicate(format: "siteID == %lld", sampleSiteID)), 9)
```

Comment:
does the test insert any tags not for the product's site? just wondering because all the other relationships don't require this site predicate

Reply:
this or a separate test case could also be a good place to assert the order of the product tags when upserting to Storage layer

Reply:
> does the test insert any tags not for the product's site? just wondering because all the other relationships don't require this site predicate

Since `ProductTag` can exist also if they are not related to a specific product, this means that if we previously add some tags, the count will be wrong.

> this or a separate test case could also be a good place to assert the order of the product tags when upserting to Storage layer

I'm implementing a separate test for this case. 👍

Reply:
Yes, we did the reset at the beginning of every test case, but in these test specific test cases, we want to demonstrate that an existing product, if edited, change their params, including tags. For example for one test case `testUpdateStoredProductEffectivelyUpdatesCorrectSitesProduct()`, we want to store a sample product, make sure that we stored 9 tags for it. We also add a new product with a different site ID, and make sure that the number of tags was not summed with the first one. In the end, we mutate the initial sample product, removing some tags, and make sure that at the ends the tags related to this product are 5. In this case, if we don't use a predicate we need to compare the value with the sum of all the tags added during this test, which is a little bit strange.